File: src/com/gfilipe/fittrack/MainActivity.java
package com.gfilipe.fittrack;

import android.app.Activity;
import android.content.SharedPreferences;
import android.graphics.Color;
import android.os.Bundle;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.TextView;

import java.util.ArrayDeque;
import java.util.Deque;
import java.util.List;

import com.gfilipe.fittrack.Models.Exercise;
import com.gfilipe.fittrack.Models.InfoItem;
import com.gfilipe.fittrack.Models.MealOption;
import com.gfilipe.fittrack.Models.MealSlot;
import com.gfilipe.fittrack.Models.Recipe;
import com.gfilipe.fittrack.Models.WorkoutDay;

public class MainActivity extends Activity {

    private static final int BG = Color.parseColor("#121212");
    private static final int CARD_BG = Color.parseColor("#1E1E1E");
    private static final int ACCENT = Color.parseColor("#E53E3E");
    private static final int TEXT_PRIMARY = Color.parseColor("#F5F5F5");
    private static final int TEXT_SECONDARY = Color.parseColor("#AAAAAA");

    private final Deque<Screen> stack = new ArrayDeque<>();
    private SharedPreferences prefs;

    private static class Screen {
        final String type;
        final String p1;
        final String p2;

        Screen(String type) {
            this(type, null, null);
        }

        Screen(String type, String p1) {
            this(type, p1, null);
        }

        Screen(String type, String p1, String p2) {
            this.type = type;
            this.p1 = p1;
            this.p2 = p2;
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        prefs = getSharedPreferences("fittrack_prefs", MODE_PRIVATE);
        stack.push(new Screen("home"));
        render();
    }

    @Override
    public void onBackPressed() {
        if (stack.size() > 1) {
            stack.pop();
            render();
        } else {
            super.onBackPressed();
        }
    }

    private void goBack() {
        onBackPressed();
    }

    private void navigate(Screen s) {
        stack.push(s);
        render();
    }

    private int dp(int value) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, value, getResources().getDisplayMetrics());
    }

    private int sp(int value) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, value, getResources().getDisplayMetrics());
    }

    // ---------------- RENDER ----------------

    private void render() {
        Screen s = stack.peek();

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setPadding(dp(16), dp(28), dp(16), dp(40));
        root.setLayoutParams(new LinearLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        if (stack.size() > 1) {
            TextView back = textView("← Voltar", 16, ACCENT, true);
            back.setPadding(0, 0, 0, dp(12));
            back.setOnClickListener(v -> goBack());
            root.addView(back);
        }

        TextView title = textView(titleFor(s), 22, TEXT_PRIMARY, true);
        title.setPadding(0, 0, 0, dp(16));
        root.addView(title);

        switch (s.type) {
            case "home":
                buildHome(root);
                break;
            case "workoutList":
                buildWorkoutList(root);
                break;
            case "workoutDetail":
                buildWorkoutDetail(root, s.p1);
                break;
            case "mealSlotList":
                buildMealSlotList(root);
                break;
            case "mealOptionList":
                buildMealOptionList(root, s.p1);
                break;
            case "mealOptionDetail":
                buildMealOptionDetail(root, s.p1, s.p2);
                break;
            case "recipeList":
                buildRecipeList(root);
                break;
            case "recipeDetail":
                buildRecipeDetail(root, s.p1);
                break;
            case "infoList":
                buildInfoList(root, s.p1);
                break;
            case "infoDetail":
                buildInfoDetail(root, s.p1, s.p2);
                break;
            default:
                break;
        }

        ScrollView scroll = new ScrollView(this);
        scroll.setBackgroundColor(BG);
        scroll.addView(root);
        setContentView(scroll);
    }

    private String titleFor(Screen s) {
        switch (s.type) {
            case "home":
                return "FitTrack — O meu plano";
            case "workoutList":
                return "Treinos";
            case "workoutDetail":
                return findWorkout(s.p1).title;
            case "mealSlotList":
                return "Refeições";
            case "mealOptionList":
                return findSlot(s.p1).title;
            case "mealOptionDetail":
                return findOption(s.p1, s.p2).name;
            case "recipeList":
                return "Receitas";
            case "recipeDetail":
                return findRecipe(s.p1).title;
            case "infoList":
                return "supplements".equals(s.p1) ? "Suplementos" : "Notas gerais";
            case "infoDetail":
                return findInfo(s.p1, s.p2).title;
            default:
                return "FitTrack";
        }
    }

    // ---------------- LOOKUPS ----------------

    private WorkoutDay findWorkout(String id) {
        for (WorkoutDay w : PlanData.WORKOUTS) if (w.id.equals(id)) return w;
        return null;
    }

    private MealSlot findSlot(String id) {
        for (MealSlot slot : PlanData.MEAL_SLOTS) if (slot.id.equals(id)) return slot;
        return null;
    }

    private MealOption findOption(String slotId, String optionId) {
        MealSlot slot = findSlot(slotId);
        if (slot == null) return null;
        for (MealOption o : slot.options) if (o.id.equals(optionId)) return o;
        return null;
    }

    private Recipe findRecipe(String id) {
        for (Recipe r : PlanData.RECIPES) if (r.id.equals(id)) return r;
        return null;
    }

    private List<InfoItem> infoListFor(String key) {
        return "supplements".equals(key) ? PlanData.SUPPLEMENTS : PlanData.GENERAL_NOTES;
    }

    private InfoItem findInfo(String key, String id) {
        for (InfoItem i : infoListFor(key)) if (i.id.equals(id)) return i;
        return null;
    }

    // ---------------- SCREENS ----------------

    private void buildHome(LinearLayout root) {
        root.addView(textView("Plano pessoal de treino e alimentação", 14, TEXT_SECONDARY, false));
        spacer(root, 12);
        addMenuItem(root, "Treinos", "Dias de treino e exercícios", () -> navigate(new Screen("workoutList")));
        addMenuItem(root, "Refeições", "Pequeno-almoço, almoço, lanche, jantar...", () -> navigate(new Screen("mealSlotList")));
        addMenuItem(root, "Receitas", "Receitas detalhadas com preparação", () -> navigate(new Screen("recipeList")));
        addMenuItem(root, "Suplementos", "Informação sobre suplementação", () -> navigate(new Screen("infoList", "supplements")));
        addMenuItem(root, "Notas gerais", "Água, refeições e regras do plano", () -> navigate(new Screen("infoList", "notes")));
    }

    private void buildWorkoutList(LinearLayout root) {
        for (WorkoutDay w : PlanData.WORKOUTS) {
            addListCard(root, w.title, w.subtitle, () -> navigate(new Screen("workoutDetail", w.id)));
        }
    }

    private void buildWorkoutDetail(LinearLayout root, String id) {
        WorkoutDay w = findWorkout(id);
        if (w == null) return;
        root.addView(textView(w.subtitle, 14, TEXT_SECONDARY, false));
        spacer(root, 12);
        for (Exercise e : w.exercises) {
            LinearLayout card = card();
            card.addView(textView(e.name, 16, TEXT_PRIMARY, true));
            TextView detail = textView(e.detail, 14, TEXT_SECONDARY, false);
            detail.setPadding(0, dp(4), 0, 0);
            card.addView(detail);
            root.addView(card);
        }
        spacer(root, 8);
        boolean done = prefs.getBoolean("workout_done_" + w.id, false);
        TextView button = textView(done ? "✓  Concluído" : "Concluir", 16, Color.WHITE, true);
        button.setGravity(Gravity.CENTER);
        button.setBackgroundColor(ACCENT);
        button.setPadding(dp(16), dp(14), dp(16), dp(14));
        button.setOnClickListener(v -> {
            boolean newDone = !prefs.getBoolean("workout_done_" + w.id, false);
            prefs.edit().putBoolean("workout_done_" + w.id, newDone).apply();
            render();
        });
        root.addView(button);
    }

    private void buildMealSlotList(LinearLayout root) {
        for (MealSlot slot : PlanData.MEAL_SLOTS) {
            addListCard(root, slot.title, slot.options.size() + " opções", () -> navigate(new Screen("mealOptionList", slot.id)));
        }
    }

    private void buildMealOptionList(LinearLayout root, String slotId) {
        MealSlot slot = findSlot(slotId);
        if (slot == null) return;
        if (slot.note != null) {
            root.addView(noteBox(slot.note));
            spacer(root, 8);
        }
        for (MealOption o : slot.options) {
            addListCard(root, o.name, null, () -> navigate(new Screen("mealOptionDetail", slot.id, o.id)));
        }
    }

    private void buildMealOptionDetail(LinearLayout root, String slotId, String optionId) {
        MealOption o = findOption(slotId, optionId);
        if (o == null) return;
        if ((o.ingredients == null || o.ingredients.isEmpty()) && o.prep == null) {
            root.addView(noteBox("Ainda sem detalhe guardado para esta opção. Podes adicionar os ingredientes mais tarde em PlanData.java."));
        }
        if (o.ingredients != null) {
            for (String ing : o.ingredients) {
                LinearLayout card = card();
                card.addView(textView(ing, 14, TEXT_PRIMARY, false));
                root.addView(card);
            }
        }
        if (o.prep != null) {
            spacer(root, 8);
            root.addView(textView("Preparação", 18, TEXT_PRIMARY, true));
            spacer(root, 4);
            root.addView(textView(o.prep, 14, TEXT_PRIMARY, false));
        }
        if (o.note != null) {
            spacer(root, 8);
            root.addView(noteBox(o.note));
        }
    }

    private void buildRecipeList(LinearLayout root) {
        for (Recipe r : PlanData.RECIPES) {
            addListCard(root, r.title, r.source, () -> navigate(new Screen("recipeDetail", r.id)));
        }
        if (!PlanData.RECIPE_TITLES_WITHOUT_DETAIL.isEmpty()) {
            spacer(root, 8);
            root.addView(textView("Sem detalhe guardado ainda", 14, TEXT_SECONDARY, true));
            spacer(root, 4);
            for (String title : PlanData.RECIPE_TITLES_WITHOUT_DETAIL) {
                LinearLayout card = card();
                card.addView(textView(title, 14, TEXT_PRIMARY, false));
                root.addView(card);
            }
        }
    }

    private void buildRecipeDetail(LinearLayout root, String id) {
        Recipe r = findRecipe(id);
        if (r == null) return;
        if (r.source != null) {
            root.addView(textView(r.source, 13, TEXT_SECONDARY, false));
            spacer(root, 8);
        }
        root.addView(textView("Ingredientes", 18, TEXT_PRIMARY, true));
        spacer(root, 4);
        for (String ing : r.ingredients) {
            root.addView(textView("•  " + ing, 14, TEXT_PRIMARY, false));
        }
        spacer(root, 12);
        root.addView(textView("Preparação", 18, TEXT_PRIMARY, true));
        spacer(root, 4);
        root.addView(textView(r.prep, 14, TEXT_PRIMARY, false));
    }

    private void buildInfoList(LinearLayout root, String key) {
        for (InfoItem item : infoListFor(key)) {
            addListCard(root, item.title, null, () -> navigate(new Screen("infoDetail", key, item.id)));
        }
    }

    private void buildInfoDetail(LinearLayout root, String key, String id) {
        InfoItem item = findInfo(key, id);
        if (item == null) return;
        root.addView(textView(item.text, 15, TEXT_PRIMARY, false));
    }

    // ---------------- WIDGET HELPERS ----------------

    private interface Action {
        void run();
    }

    private TextView textView(String text, int spSize, int color, boolean bold) {
        TextView tv = new TextView(this);
        tv.setText(text);
        tv.setTextSize(spSize);
        tv.setTextColor(color);
        if (bold) tv.setTypeface(tv.getTypeface(), android.graphics.Typeface.BOLD);
        return tv;
    }

    private void spacer(LinearLayout root, int heightDp) {
        View v = new View(this);
        v.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, dp(heightDp)));
        root.addView(v);
    }

    private LinearLayout card() {
        LinearLayout card = new LinearLayout(this);
        card.setOrientation(LinearLayout.VERTICAL);
        card.setBackgroundColor(CARD_BG);
        card.setPadding(dp(14), dp(12), dp(14), dp(12));
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        lp.setMargins(0, 0, 0, dp(8));
        card.setLayoutParams(lp);
        return card;
    }

    private void addListCard(LinearLayout root, String title, String subtitle, Action onClick) {
        LinearLayout card = card();
        card.setClickable(true);
        card.setFocusable(true);
        card.addView(textView(title, 16, TEXT_PRIMARY, true));
        if (subtitle != null) {
            TextView sub = textView(subtitle, 13, TEXT_SECONDARY, false);
            sub.setPadding(0, dp(4), 0, 0);
            card.addView(sub);
        }
        card.setOnClickListener(v -> onClick.run());
        root.addView(card);
    }

    private void addMenuItem(LinearLayout root, String title, String subtitle, Action onClick) {
        addListCard(root, title, subtitle, onClick);
    }

    private LinearLayout noteBox(String text) {
        LinearLayout box = new LinearLayout(this);
        box.setOrientation(LinearLayout.VERTICAL);
        box.setBackgroundColor(Color.parseColor("#332222"));
        box.setPadding(dp(14), dp(12), dp(14), dp(12));
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        lp.setMargins(0, 0, 0, dp(8));
        box.setLayoutParams(lp);
        box.addView(textView(text, 14, TEXT_PRIMARY, false));
        return box;
    }
}

File: app/src/main/java/com/gfilipe/fittrack/MainActivity.kt
package com.gfilipe.fittrack

import android.content.Context
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Fastfood
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.MenuBook
import androidx.compose.material.icons.filled.RadioButtonUnchecked
import androidx.compose.material.icons.filled.Science
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Button
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import androidx.navigation.compose.NavHost
import androidx.navigation.compose.composable
import androidx.navigation.compose.rememberNavController
import com.gfilipe.fittrack.data.InfoItem
import com.gfilipe.fittrack.data.MealOption
import com.gfilipe.fittrack.data.MealSlot
import com.gfilipe.fittrack.data.PlanData
import com.gfilipe.fittrack.data.Recipe
import com.gfilipe.fittrack.data.WorkoutDay
import com.gfilipe.fittrack.ui.theme.FitTrackTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            FitTrackTheme {
                Surface(modifier = Modifier.fillMaxSize()) {
                    FitTrackApp()
                }
            }
        }
    }
}

private fun prefs(context: Context) = context.getSharedPreferences("fittrack_prefs", Context.MODE_PRIVATE)

@Composable
fun FitTrackApp() {
    val navController = rememberNavController()
    NavHost(navController = navController, startDestination = "home") {
        composable("home") { HomeScreen(navController) }

        composable("workouts") { WorkoutListScreen(navController) }
        composable("workout/{id}") { backStackEntry ->
            val id = backStackEntry.arguments?.getString("id")
            val workout = PlanData.workouts.find { it.id == id }
            if (workout != null) WorkoutDetailScreen(navController, workout)
        }

        composable("meals") { MealSlotListScreen(navController) }
        composable("meals/{slotId}") { backStackEntry ->
            val slotId = backStackEntry.arguments?.getString("slotId")
            val slot = PlanData.mealSlots.find { it.id == slotId }
            if (slot != null) MealOptionListScreen(navController, slot)
        }
        composable("meals/{slotId}/{optionId}") { backStackEntry ->
            val slotId = backStackEntry.arguments?.getString("slotId")
            val optionId = backStackEntry.arguments?.getString("optionId")
            val slot = PlanData.mealSlots.find { it.id == slotId }
            val option = slot?.options?.find { it.id == optionId }
            if (slot != null && option != null) MealOptionDetailScreen(navController, slot, option)
        }

        composable("recipes") { RecipeListScreen(navController) }
        composable("recipe/{id}") { backStackEntry ->
            val id = backStackEntry.arguments?.getString("id")
            val recipe = PlanData.recipes.find { it.id == id }
            if (recipe != null) RecipeDetailScreen(navController, recipe)
        }

        composable("supplements") { InfoListScreen(navController, "Suplementos", PlanData.supplements, "supplement") }
        composable("supplement/{id}") { backStackEntry ->
            val id = backStackEntry.arguments?.getString("id")
            val item = PlanData.supplements.find { it.id == id }
            if (item != null) InfoDetailScreen(navController, item)
        }

        composable("notes") { InfoListScreen(navController, "Notas gerais", PlanData.generalNotes, "note") }
        composable("note/{id}") { backStackEntry ->
            val id = backStackEntry.arguments?.getString("id")
            val item = PlanData.generalNotes.find { it.id == id }
            if (item != null) InfoDetailScreen(navController, item)
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun FitTrackScaffold(
    title: String,
    navController: NavController,
    showBack: Boolean = true,
    content: @Composable (PaddingValues) -> Unit
) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(title, fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    if (showBack) {
                        IconButton(onClick = { navController.popBackStack() }) {
                            Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Voltar")
                        }
                    }
                }
            )
        }
    ) { padding -> content(padding) }
}

@Composable
private fun SectionCard(title: String, subtitle: String? = null, onClick: () -> Unit) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 6.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant),
        onClick = onClick
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(title, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
            if (subtitle != null) {
                Text(
                    subtitle,
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(top = 4.dp),
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

// ---------------- HOME ----------------

@Composable
fun HomeScreen(navController: NavController) {
    FitTrackScaffold(title = "FitTrack — O meu plano", navController = navController, showBack = false) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            item {
                Text(
                    "Plano pessoal de treino e alimentação",
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            item {
                HomeMenuItem(Icons.Filled.FitnessCenter, "Treinos", "Dias de treino e exercícios") {
                    navController.navigate("workouts")
                }
            }
            item {
                HomeMenuItem(Icons.Filled.Fastfood, "Refeições", "Pequeno-almoço, almoço, lanche, jantar…") {
                    navController.navigate("meals")
                }
            }
            item {
                HomeMenuItem(Icons.Filled.MenuBook, "Receitas", "Receitas detalhadas com preparação") {
                    navController.navigate("recipes")
                }
            }
            item {
                HomeMenuItem(Icons.Filled.Science, "Suplementos", "Informação sobre suplementação") {
                    navController.navigate("supplements")
                }
            }
            item {
                HomeMenuItem(Icons.Filled.Info, "Notas gerais", "Água, refeições e regras do plano") {
                    navController.navigate("notes")
                }
            }
        }
    }
}

@Composable
private fun HomeMenuItem(icon: androidx.compose.ui.graphics.vector.ImageVector, title: String, subtitle: String, onClick: () -> Unit) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 6.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant),
        onClick = onClick
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Icon(icon, contentDescription = null, tint = MaterialTheme.colorScheme.primary, modifier = Modifier.size(32.dp))
            Column {
                Text(title, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
                Text(subtitle, style = MaterialTheme.typography.bodySmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
            }
        }
    }
}

// ---------------- TREINOS ----------------

@Composable
fun WorkoutListScreen(navController: NavController) {
    FitTrackScaffold(title = "Treinos", navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            items(PlanData.workouts) { workout ->
                SectionCard(title = workout.title, subtitle = workout.subtitle) {
                    navController.navigate("workout/${workout.id}")
                }
            }
        }
    }
}

@Composable
fun WorkoutDetailScreen(navController: NavController, workout: WorkoutDay) {
    val context = LocalContext.current
    var done by remember {
        mutableStateOf(prefs(context).getBoolean("workout_done_${workout.id}", false))
    }
    FitTrackScaffold(title = workout.title, navController = navController) { padding ->
        Column(modifier = Modifier.padding(padding)) {
            Text(
                workout.subtitle,
                style = MaterialTheme.typography.bodyMedium,
                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            LazyColumn(modifier = Modifier.weight(1f), contentPadding = PaddingValues(bottom = 12.dp)) {
                items(workout.exercises) { exercise ->
                    Card(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 16.dp, vertical = 6.dp),
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
                    ) {
                        Column(modifier = Modifier.padding(16.dp)) {
                            Text(exercise.name, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
                            Text(
                                exercise.detail,
                                style = MaterialTheme.typography.bodyMedium,
                                modifier = Modifier.padding(top = 4.dp),
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }
            }
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                horizontalArrangement = Arrangement.Center
            ) {
                Button(onClick = {
                    done = !done
                    prefs(context).edit().putBoolean("workout_done_${workout.id}", done).apply()
                }) {
                    Icon(
                        if (done) Icons.Filled.CheckCircle else Icons.Filled.RadioButtonUnchecked,
                        contentDescription = null
                    )
                    Text(if (done) "  Concluído" else "  Concluir", modifier = Modifier.padding(start = 4.dp))
                }
            }
        }
    }
}

// ---------------- REFEIÇÕES ----------------

@Composable
fun MealSlotListScreen(navController: NavController) {
    FitTrackScaffold(title = "Refeições", navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            items(PlanData.mealSlots) { slot ->
                SectionCard(title = slot.title, subtitle = "${slot.options.size} opções") {
                    navController.navigate("meals/${slot.id}")
                }
            }
        }
    }
}

@Composable
fun MealOptionListScreen(navController: NavController, slot: MealSlot) {
    FitTrackScaffold(title = slot.title, navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            if (slot.note != null) {
                item {
                    NoteBox(slot.note)
                }
            }
            items(slot.options) { option ->
                SectionCard(title = option.name) {
                    navController.navigate("meals/${slot.id}/${option.id}")
                }
            }
        }
    }
}

@Composable
fun MealOptionDetailScreen(navController: NavController, slot: MealSlot, option: MealOption) {
    FitTrackScaffold(title = option.name, navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            if (option.ingredients.isEmpty() && option.prep == null) {
                item {
                    NoteBox("Ainda sem detalhe guardado para esta opção. Podes adicionar os ingredientes mais tarde em PlanData.kt.")
                }
            }
            items(option.ingredients) { ingredient ->
                Card(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 16.dp, vertical = 4.dp),
                    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
                ) {
                    Text(ingredient.text, modifier = Modifier.padding(12.dp), style = MaterialTheme.typography.bodyMedium)
                }
            }
            if (option.prep != null) {
                item {
                    Text(
                        "Preparação",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp)
                    )
                    Text(
                        option.prep,
                        style = MaterialTheme.typography.bodyMedium,
                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp)
                    )
                }
            }
            if (option.note != null) {
                item { NoteBox(option.note) }
            }
        }
    }
}

// ---------------- RECEITAS ----------------

@Composable
fun RecipeListScreen(navController: NavController) {
    FitTrackScaffold(title = "Receitas", navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            items(PlanData.recipes) { recipe ->
                SectionCard(title = recipe.title, subtitle = recipe.source) {
                    navController.navigate("recipe/${recipe.id}")
                }
            }
            if (PlanData.recipeTitlesWithoutDetail.isNotEmpty()) {
                item {
                    Text(
                        "Sem detalhe guardado ainda",
                        style = MaterialTheme.typography.titleSmall,
                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
                items(PlanData.recipeTitlesWithoutDetail) { title ->
                    Card(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 16.dp, vertical = 4.dp)
                    ) {
                        Text(title, modifier = Modifier.padding(12.dp))
                    }
                }
            }
        }
    }
}

@Composable
fun RecipeDetailScreen(navController: NavController, recipe: Recipe) {
    FitTrackScaffold(title = recipe.title, navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            if (recipe.source != null) {
                item {
                    Text(
                        recipe.source,
                        style = MaterialTheme.typography.bodySmall,
                        fontStyle = androidx.compose.ui.text.font.FontStyle.Italic,
                        modifier = Modifier.padding(horizontal = 16.dp, bottom = 8.dp),
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
            item {
                Text(
                    "Ingredientes",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp)
                )
            }
            items(recipe.ingredients) { ingredient ->
                Text("•  $ingredient", modifier = Modifier.padding(horizontal = 16.dp, vertical = 2.dp))
            }
            item {
                Text(
                    "Preparação",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.padding(horizontal = 16.dp, top = 16.dp, bottom = 4.dp)
                )
                Text(recipe.prep, modifier = Modifier.padding(horizontal = 16.dp))
            }
        }
    }
}

// ---------------- INFO (Suplementos / Notas) ----------------

@Composable
fun InfoListScreen(navController: NavController, title: String, infoItems: List<InfoItem>, route: String) {
    FitTrackScaffold(title = title, navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(vertical = 12.dp), modifier = Modifier.padding(padding)) {
            items(infoItems) { item ->
                SectionCard(title = item.title) {
                    navController.navigate("$route/${item.id}")
                }
            }
        }
    }
}

@Composable
fun InfoDetailScreen(navController: NavController, item: InfoItem) {
    FitTrackScaffold(title = item.title, navController = navController) { padding ->
        LazyColumn(contentPadding = PaddingValues(16.dp), modifier = Modifier.padding(padding)) {
            item { Text(item.text, style = MaterialTheme.typography.bodyLarge) }
        }
    }
}

@Composable
private fun NoteBox(text: String) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 8.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.12f))
    ) {
        Text(text, modifier = Modifier.padding(12.dp), style = MaterialTheme.typography.bodyMedium)
    }
}
